### ISSUE TITLE
Error builing app

### ISSUE THREAD

**Issue by joenuts69** [Natoune/SpotifyMobileLyricsAPI#27]
![image](https://github.com/user-attachments/assets/520fd464-02cb-4aa6-bfcf-5c12983abade)
I'm not sure what the issue is here

**Comment by Natoune** [comment]
You need to clone the repository (`git clone https://github.com/Natoune/SpotifyMobileLyricsAPI.git` or download and extract the [SpotifyMobileLyricsAPI-main.zip](https://github.com/Natoune/SpotifyMobileLyricsAPI/archive/refs/heads/main.zip) file).
Then run the command `npm run patch:xmanager ...` IN the cloned repository folder.

**Comment by joenuts69** [comment]
![Screenshot_20241212_144901_WhatsApp.jpg](https://github.com/user-attachments/assets/2d8f4b3c-e491-419c-ae50-96e674d244ac)

This is the issue I am seeing when keytool is installed and I just used it to make a key

**Comment by Natoune** [comment]
Is keytool present in your system path variable?

**Comment by joenuts69** [comment]
Yes it is. As I said I already made a key so I know it's configed properly

**Comment by Natoune** [comment]
Re-clone the repo to have the latest patch version and manually pass your keystore created with keytool to the script with the `--ks-file` and `--ks-pass` arguments.

**Comment by joenuts69** [comment]
I have another error, I have made sure that the path and password of the keyfile is correct:

Building patched APK...
Signing APK...
node:internal/errors:983
  const err = new Error(message);
              ^

Error: Command failed: .\build-tools\apksigner.bat sign --ks "keystore.jks" --ks-pass "pass:thepassword" --out ..\Patched.apk aligned.apk
    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at checkExecSyncError (node:child_process:882:11)
    at execSync (node:child_process:954:15)
    at C:\Users\Guy\Documents\spotify\SpotifyMobileLyricsAPI\scripts\xmanager.patch.js:653:3
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5) {
  status: 2,
  signal: null,
  output: [ null, null, null ],
  pid: 6124,
  stdout: null,
  stderr: null
}

Node.js v23.4.0

C:\Users\Guy\Documents\spotify\SpotifyMobileLyricsAPI>

**Comment by Natoune** [comment]
Hey, sorry for the late reply
The latest commit may fix this error. Try downloading the [latest version of the script](https://raw.githubusercontent.com/Natoune/SpotifyMobileLyricsAPI/refs/heads/main/scripts/xmanager.patch.js) and running it again.

**Comment by joenuts69** [comment]
I got the app built and was able to login but when I try to play a song through Spotify Connect, the app crashes https://github.com/user-attachments/assets/b6b3b323-5335-4d22-a2fb-df238658f35c

**Comment by Natoune** [comment]
Do you need this version specifically? The patch is only tested on a few versions. The latest functional tested version for the patch is the v8.9.98.488. You can patch this APK using the script to have this version: [Spotify v8.9.98.488 [AB Experimental] (Merged).apk](https://cdn.discordapp.com/attachments/1076920612624281670/1314881752660971610/Spotify_v8.9.98.488_AB_Experimental_Merged.apk?ex=6772638f&is=6771120f&hm=79959917be1f91d005657b1f0b02111a0ab0a27b3647b25a1342c3c3ed0ebcdd&)

**Comment by joenuts69** [comment]
It doesn't really matter which version it works on, once it works. The version that you provided isn't working as I am still limited to the free features such as shuffling.

**Comment by Natoune** [comment]
The version I sent should work. I only deal with the lyrics patch, if other features don't work, you have to check with the xManager team...
